tests/test_fields_mitigate_collision.py

```diff
@@ -27,10 +27,40 @@ class TestMessage(proto.Message):
         spam_ = proto.Field(proto.STRING, number=1)
         eggs = proto.Field(proto.STRING, number=2)
 
+    class TextStream(proto.Message):
+        text_stream = proto.Field(TestMessage, number=1)
+
     obj = TestMessage(spam_="has_spam")
     obj.eggs = "has_eggs"
     assert obj.spam_ == "has_spam"
 
     # Test that `spam` is coerced to `spam_`
     modified_obj = TestMessage({"spam": "has_spam", "eggs": "has_eggs"})
     assert modified_obj.spam_ == "has_spam"
+
+    # Test __setattr__ and __getattr___
+    modified_obj.__setattr__("spam", "no_spam")
+    modified_obj.__getattr__("spam") == "no_spam"
+
+    modified_obj.__setattr__("spam_", "yes_spam")
+    modified_obj.__getattr__("spam_") == "yes_spam"
```

Comment:
This works, but it's more idiomatic and arguably a better test to just try to interact with the attributes directly:
```python
modified_obj.spam = "no_spam"
assert modified_obj.spam == "no_spam"

modified_obj.spam_ = "yes_spam"
assert modified_obj.spam_ == "yes_spam"
```
It may be a good idea to go bidirectionally too:
```python
modified_obj.spam = "maybe_spam"
assert modified_obj.spam_ == "maybe_spam"

modified_obj.spam_ = "maybe_not_spam"
assert modified_obj.spam == "maybe_not_spam"
```